tiledb/sm/c_api/tiledb.cc

```diff
@@ -3675,7 +3675,7 @@ int32_t tiledb_subarray_set_config(
 
 void tiledb_subarray_free(tiledb_subarray_t** subarray) {
   if (subarray != nullptr && *subarray != nullptr) {
-    delete (*subarray)->subarray_;
+    (*subarray)->subarray_ = nullptr;
```

Comment:
Calling `reset()` is more efficient that create a new `shared_ptr` and assigning it.

Reply:
`subarray_` is not a nullptr; it's a `Subarray*`